python/interpret_community/explanation/explanation.py

```diff
@@ -1666,145 +1669,161 @@ def get_value(x):
+        if hasattr(explanation, prop) and getattr(explanation, prop) is not None:
+            value = getattr(explanation, prop)
             if isinstance(value, pd.DataFrame):
-                expldict[param] = value.values.tolist()
-                _metadata[param] = 'DataFrame'
+                value = value.values.tolist()
+                metadata = 'DataFrame'
             elif isinstance(value, DatasetWrapper):
-                expldict[param] = value.original_dataset.tolist()
-                _metadata[param] = 'DatasetWrapper'
+                value = value.original_dataset.tolist()
+                metadata = 'DatasetWrapper'
             elif isinstance(value, DenseData):
-                expldict[param] = value.original_dataset.tolist()
-                _metadata[param] = 'DenseData'
+                value = value.original_dataset.tolist()
+                metadata = 'DenseData'
             elif isinstance(value, np.ndarray):
-                expldict[param] = value.tolist()
-                _metadata[param] = 'ndarray'
+                value = value.tolist()
+                metadata = 'ndarray'
+            elif isinstance(value, list):
+                metadata = 'list'
             else:
-                expldict[param] = value
-    return json.dumps({
-        '_metadata': _metadata,
-        'explanation': expldict
-    })
-
+                metadata = 'other'
+            data_dict = {
+                'metadata': metadata,
+                'data': value
+            }
+            filename = os.path.join(path, prop + '.json')
+            with open(filename, 'w') as f:
+                json.dump(data_dict, f)
+    # create metadata file
+    classification = ClassesMixin._does_quack(explanation)
+    is_raw = False if not FeatureImportanceExplanation._does_quack(explanation) else explanation.is_raw
+    is_eng = False if not FeatureImportanceExplanation._does_quack(explanation) else explanation.is_engineered
+    feature_quack = FeatureImportanceExplanation._does_quack
+    prop_dict = {
+        ExplainParams.EXPLANATION_ID: explanation.id,
+        ExplainType.MODEL: ExplainType.CLASSIFICATION if classification else ExplainType.REGRESSION,
+        ExplainType.DATA: ExplainType.TABULAR,
+        ExplainType.MODEL_TASK: explanation.model_task,
+        ExplainType.METHOD: explanation.method,
+        ExplainType.MODEL_CLASS: explanation.model_type,
+        ExplainType.IS_RAW: is_raw,
+        ExplainType.IS_ENG: is_eng,
+        ExplainType.GLOBAL: GlobalExplanation._does_quack(explanation),
+        ExplainType.LOCAL: LocalExplanation._does_quack(explanation),
+        ExplainParams.NUM_CLASSES: 1 if not ClassesMixin._does_quack(explanation) else explanation.num_classes,
+        ExplainParams.NUM_EXAMPLES: 0 if not LocalExplanation._does_quack(explanation) else explanation.num_examples,
+        ExplainParams.NUM_FEATURES: 0 if not feature_quack(explanation) else explanation.num_features,
```

Comment:
nit: instead of making an alias for the method, just make a variable for the computation:

```
num_features = 0 if not feature_quack(explanation) else explanation.num_features
...
    ExplainParams.NUM_FEATURES: num_features
```